lightningd/gossip/gossip.c

```diff
@@ -418,15 +418,27 @@ static struct io_plan *release_peer(struct io_conn *conn, struct daemon *daemon,
 			      "%s", tal_hex(trc, msg));
 
 	peer = find_peer(daemon, unique_id);
-	if (!peer)
-		status_failed(WIRE_GOSSIPSTATUS_BAD_RELEASE_REQUEST,
-			      "Unknown peer %"PRIu64, unique_id);
+	if (!peer) {
+		/* This can happen with a reconnect vs connect race.
+		 * See gossip_peer_released in master daemon. */
+		struct crypto_state dummy;
```

Comment:
Yeah, we don't have a way of handling multiple types of replies to a single request.  If it happens elsewhere too, we might consider getting something more generic...